fs/metrics.h

```diff
@@ -2,13 +2,134 @@
+    } else if (type_string == type_count_reporter) {
+      AddReporter(label, Reporter(factory_->BuildCountReporter(meta_alloc_qps_label, bytedance_tags_), type_string));
+    } else {
+      assert(false);
+      AddReporter(label, Reporter(nullptr, ""));
+    }
+  }
+
+public: // For Record Reporter:
+  virtual void AddRecord(const std::string& label, size_t size) override {
+    assert(reporter_map_.find(label) != reporter_map_.end());
+    Reporter rep = reporter_map_[label];
+    if (rep.type_ == type_hist_reporter) {
+      reinterpret_cast<HistReporterHandle*>(rep.handle_)->AddRecord(size);
+    } else if (rep.type_ == type_count_reporter) {
+      reinterpret_cast<CountReporterHandle*>(rep.handle_)->AddCount(size);
+    } else {
+      assert(false);
+    }
+  }
+  virtual void* GetRecord(const std::string& label) override {
+    assert(false);
+    return reporter_map_[label].handle_;
+  }
+    
+  virtual ~BDZenFSMetrics() {}
+
+ public:
+  BDZenFSMetrics(std::shared_ptr<MetricsReporterFactory> factory, std::string bytedance_tags, std::shared_ptr<Logger> logger):
+    ZenFSMetrics(),
+    bytedance_tags_(bytedance_tags),
+    factory_(new CurriedMetricsReporterFactory(factory, logger.get(), Env::Default())) {
+      AddReporter(fg_write_lat_label, type_hist_reporter);
+      AddReporter(bg_write_lat_label, type_hist_reporter);
+      AddReporter(read_lat_label, type_hist_reporter);
+      AddReporter(fg_sync_lat_label, type_hist_reporter);
+      AddReporter(bg_sync_lat_label, type_hist_reporter);
+      AddReporter(meta_alloc_lat_label, type_hist_reporter);
+      AddReporter(sync_metadata_lat_label, type_hist_reporter);
+      AddReporter(io_alloc_wal_lat_label, type_hist_reporter);
+      AddReporter(io_alloc_wal_actual_lat_label, type_hist_reporter);
+      AddReporter(io_alloc_non_wal_lat_label, type_hist_reporter);
+      AddReporter(io_alloc_non_wal_actual_lat_label, type_hist_reporter);
+      AddReporter(roll_lat_label, type_hist_reporter);
+
+      AddReporter(write_qps_label, type_count_reporter);
+      AddReporter(read_qps_label, type_count_reporter);
+      AddReporter(sync_qps_label, type_count_reporter);
+      AddReporter(meta_alloc_qps_label, type_count_reporter);
+      AddReporter(io_alloc_qps_label, type_count_reporter);
+      AddReporter(roll_qps_label, type_count_reporter);
+      AddReporter(write_throughput_label, type_count_reporter);
+      AddReporter(roll_throughput_label, type_count_reporter);
+
+      AddReporter(active_zones_label, type_hist_reporter);
+      AddReporter(open_zones_label, type_hist_reporter);
+      AddReporter(zbd_free_space_label, type_hist_reporter);
+      AddReporter(zbd_used_space_label, type_hist_reporter);
+      AddReporter(zbd_reclaimable_space_label, type_hist_reporter);
+      AddReporter(zbd_resetable_zones_label, type_hist_reporter);
+    }
+
+public:
+  std::string type_hist_reporter = "zenfs_type_hist_reporter";
```

Comment:
Why not const string, or even const char*?

Reply:
> 

I had the same confusion when I saw the way the original class constants were defined, and also considered whether I should rewrite this part of the code in a reasonable way. But this part of the code really doesn't affect performance significantly, which is why I chose to keep the same format as the previous constant definition code (even if it looks weird).
I will rewrite this part of the code if necessary, as you suggest.